src/components/Loader/index.js

```diff
@@ -114,36 +114,33 @@ const dataFetchReducer = (state, action) => {
  * @param {*} updateProps Similar to `useEffect` dependencies, you can add any values here that invalidate the result of `fetch`
  */
 export function useLoader(fetch, updateProps = []) {
-  // Using a reducer helps remove the `state` dependency from the effect below
-  // React guarantees that `dispatch` is unique accross renders.
-  // https://overreacted.io/a-complete-guide-to-useeffect/#why-usereducer-is-the-cheat-mode-of-hooks
-  const [state, dispatch] = React.useReducer(dataFetchReducer, {
-    error: null,
-    isLoading: true,
-    data: null,
-  });
-
   const fetchDataCallback = React.useCallback(
     async function fetchData() {
       dispatch({ type: 'FETCH_INIT' });
-
       try {
         const data = await fetch();
         dispatch({ type: 'FETCH_SUCCESS', payload: data });
       } catch (error) {
         dispatch({ type: 'FETCH_FAILURE', payload: error });
       }
     },
-    [fetch]
+    [fetch, ...updateProps] // eslint-disable-line react-hooks/exhaustive-deps
   );
+  // Using a reducer helps remove the `state` dependency from the effect below
+  // React guarantees that `dispatch` is unique accross renders.
+  // https://overreacted.io/a-complete-guide-to-useeffect/#why-usereducer-is-the-cheat-mode-of-hooks
+  const [state, dispatch] = React.useReducer(dataFetchReducer, {
+    refresh: fetchDataCallback,
+    isLoading: true,
+    error: null,
+    hasError: false,
+    data: null,
+  });
 
+  // Call fetch when properties change.
   React.useEffect(() => {
     fetchDataCallback();
-  }, [fetchDataCallback, ...updateProps]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [...updateProps]); // eslint-disable-line react-hooks/exhaustive-deps
 
-  return {
-    ...state,
-    hasError: !!state.error,
-    refresh: fetchDataCallback,
-  };
+  return state;
```

Comment:
Any advantage to returning `state` here? seems unnecessary `hasError` and `refresh` as part of the state

Reply:
Yes, you are correct. This is has no impact on functionality. However, I thought the advantage of declaring ```hasError``` and ```refresh``` as part of the default state was improved readability. There would be no side effects if this was undone.

Reply:
I see, I was a bit confused with the callback since the state references it and the callback uses `dispatch` defined below. Maybe we can leave `hasError` in the state and move `refresh` out? 

With redux, they [recommend not having functions in the state](https://redux.js.org/faq/organizing-state#can-i-put-functions-promises-or-other-non-serializable-items-in-my-store-state). Not sure if there are any drawbacks with having them in `useReducer` hook?